core/indexer/elasticsearch_test.go

```diff
@@ -284,6 +304,35 @@ func TestElasticIndexer_buildTransactionBulks(t *testing.T) {
 	}
 }
 
+func TestElasticIndexer_buildTransactionBulksWithSCResults(t *testing.T) {
+	ei := indexer.NewTestElasticIndexer(url, username, password, shardCoordinator, marshalizer, hasher, log, &indexer.Options{})
+	testSCKey := "utx1"
+	testSCResult := &smartContractResult.SmartContractResult{
+		Nonce:  1,
+		TxHash: []byte("tx1"),
+	}
+	header := newTestBlockHeader()
+	body := newTestBlockBodyWithSc(testSCKey)
+
+	txPool := newTestTxPoolWithScResults(testSCKey, testSCResult)
+
+	bulks := ei.BuildTransactionBulks(body, header, txPool)
+
+	foundSc := false
+	for _, bulk := range bulks {
+		for _, tx := range bulk {
+			if tx.Hash == hex.EncodeToString([]byte(testSCKey)) {
+				foundSc = true
+				break
+			}
+		}
+		if foundSc {
```

Comment:
This was a trick because we don't have `break X` in go. It breaks out of the first loop if foundSc will be set to true in the second loop. And no, it cannot be changed, it is a local variable for this test, declared on line 321

Reply:
yes. My bad